src/patternfly/demos/Toolbar/examples/toolbar-complex-example.hbs

```diff
@@ -0,0 +1,59 @@
+{{#> toolbar}}
+  {{#> toolbar-section pf-l-toolbar--modifier="pf-u-justify-content-space-between pf-u-mx-xl pf-u-my-md"}}
+    {{#> toolbar-group}}
+      {{#> toolbar-item pf-l-toolbar__item--modifier="pf-u-mr-2xl"}}
+        {{#> input-group }}
+          {{#> dropdown id="dropdown-example-collapsed" pf-c-dropdown--HasToggleIcon="true"}}
+            All
+          {{/dropdown}}
+          {{#> form-control controlType="input" input="true" type="text" id="textInput10" name="textInput10"  placeholder="Filter..." aria-label="filter input with popup button"}}
+          {{/form-control}}
+          {{#> button btnClass="pf-m-tertiary" btnAttributes='id="textAreaButton2"'}} <i class="fas fa-search aria-hidden="true"></i> {{/button}}
+        {{/input-group}}
+      {{/toolbar-item}}
+      {{#> toolbar-item pf-l-toolbar__item--modifier="pf-u-mr-lg"}}
+        {{#> dropdown id="dropdown-example-collapsed" pf-c-dropdown--HasToggleIcon="true"}}
+          All
+        {{/dropdown}}
+      {{/toolbar-item}}
+      {{#> toolbar-item}}
+        {{#> button btnClass="pf-c-button pf-m-plain" btnAttributes="aria-label=\"Sort A-Z\" "}}
+          <i class="fas fa-sort-alpha-down" aria-hidden="true"></i>
+        {{/button}}
+      {{/toolbar-item}}
+    {{/toolbar-group}}
+    {{#> toolbar-group}}
+      {{#> toolbar-item pf-l-toolbar__item--modifier="pf-u-mr-md"}}
+        {{#> button btnClass="pf-c-button pf-m-plain" btnAttributes="aria-label=\"Table\" "}}
+          <i class="fas fa-table" aria-hidden="true"></i>
+        {{/button}}
+      {{/toolbar-item}}
+      {{#> toolbar-item}}
+          {{#> button btnClass="pf-c-button pf-m-plain" btnAttributes="aria-label=\"List\" "}}
+            <i class="fas fa-list" aria-hidden="true"></i>
+        {{/button}}
+      {{/toolbar-item}}
+      {{#> toolbar-item pf-l-toolbar__item--modifier="pf-u-mx-md"}}
+        {{#> button btnClass="pf-c-button pf-m-plain"}}
+          Action
+        {{/button}}
+      {{/toolbar-item}}
+      {{#> toolbar-item pf-l-toolbar__item--modifier="pf-u-mr-md"}}
+        {{#> button btnClass="pf-c-button pf-m-primary"}}
+          Action
+        {{/button}}
+      {{/toolbar-item}}
+      {{#> toolbar-item}}
+        {{#> dropdown pf-c-dropdown__kebab-icon="true" pf-c-dropdown--Modifier="pf-m-plain"}}
+        {{/dropdown}}
+      {{/toolbar-item}}
+    {{/toolbar-group}}
+  {{/toolbar-section}}
+  {{#> toolbar-section pf-l-toolbar--modifier="pf-u-justify-content-space-between pf-u-mx-xl pf-u-my-md"}}
```

Comment:
do we need the y-axis utility classes on the sections in this demo? Can we rely on the default spacing between toolbar-sections to create that space?

and is the x-axis margin just for the purposes of the demo, or is this something we would look to recommend to folks incorporating a toolbar layout?

I might drop them both, but particularly the y-axis ones so it's clear that you don't need a utility class to create space between toolbar-sections. But if they're there by design, ignore my comment :)

Reply:
Leaving the x and y axis utility classes, so that it follows the design doc, but we should look into changing this in the toolbar layout.